grapher/stackedCharts/MarimekkoChart.tsx

```diff
@@ -0,0 +1,1651 @@
+                    bars: excludeUndefined(
+                        series.map((series): Bar | undefined => {
+                            const point = series.points.find(
+                                (point) => point.position === entityName
+                            )
+                            if (!point) return undefined
+                            return {
+                                kind: BarShape.Bar,
+                                yPoint: point,
+                                color: series.color,
+                                seriesName: series.seriesName,
+                            }
+                        })
+                    ),
+                }
+            })
+            .filter((item) => item) as Item[]
+
+        return items
+    }
+
+    @computed private get sortedItems(): Item[] {
+        const { items, sortConfig } = this
+
+        let sortByFunc: (item: Item) => number | string
+        switch (sortConfig.sortBy) {
+            case SortBy.entityName:
+                sortByFunc = (item: Item): string => item.entityName
+                break
+            case SortBy.column:
+                const sortColumnSlug = sortConfig.sortColumnSlug
+                sortByFunc = (item: Item): number =>
+                    item.bars.find((b) => b.seriesName === sortColumnSlug)
+                        ?.yPoint.value ?? 0
+                break
+            default:
+            case SortBy.total:
+                sortByFunc = (item: Item): number => {
+                    const lastPoint = last(item.bars)?.yPoint
+                    if (!lastPoint) return 0
+                    return lastPoint.valueOffset + lastPoint.value
+                }
+        }
+        const sortedItems = sortBy(items, sortByFunc)
+        const sortOrder = sortConfig.sortOrder ?? SortOrder.desc
+        if (sortOrder === SortOrder.desc) sortedItems.reverse()
+
+        const [itemsWithValues, itemsWithoutValues] = partition(
+            sortedItems,
+            (item) => item.bars.length !== 0
+        )
+
+        return [...itemsWithValues, ...itemsWithoutValues]
+    }
+
+    @computed get placedItems(): PlacedItem[] {
+        const { sortedItems, dualAxis, x0, xDomainCorrectionFactor } = this
+        const placedItems: PlacedItem[] = []
+        let currentX = 0
+        for (const item of sortedItems) {
+            placedItems.push({ ...item, xPosition: currentX })
+            currentX += Math.max(
+                1,
+                dualAxis.horizontalAxis.place(
+                    item.xPoint.value * xDomainCorrectionFactor
+                ) - dualAxis.horizontalAxis.place(x0)
+            )
+        }
+        return placedItems
+    }
+
+    @computed get placedItemsMap(): Map<string, PlacedItem> {
+        return new Map(this.placedItems.map((item) => [item.entityName, item]))
+    }
+
+    // legend props
+
+    @computed get legendPaddingTop(): number {
+        return this.baseFontSize
+    }
```

Comment:
@danyx23 this is adding padding between "select countries" button and the legend.